### ISSUE TITLE
Windows service scripts runs themselves in infinite loops if the Server service is not running

### ISSUE THREAD

**Issue by Zero3** [DNSCrypt/dnscrypt-proxy#336]
`service-install.bat`, `service-restart.bat` and `service-uninstall.bat` runs themselves in infinite loops if the Server system service is not running. The Server service is responsible for file and printer sharing (aka SMB) in Windows. It is running by default, but it is not unusual to disable this service if you are careful about privacy and security and do not need file and printer sharing.

If the service is not running, these scripts then work as denial-of-service attacks against yourself because of the endless stream of command promts that pop up when you run them.

The scripts fail when the service is not running because of this hack that they use to detect whether they are missing elevated permissions:

````
@NET session
@IF %ERRORLEVEL% NEQ 0 GOTO ELEVATE
````

The `net session` command depend on the Server service, and will fail with the following error (and error level `2`) when the service is not running, regardless of whether the process has elevated permissions:

> The Server service is not started.
>
> More help is available by typing NET HELPMSG 2114.

There are many different hacks for detecting the lack of elevated permissions. I tried the following one instead, which works for fine for me on Windows 7, and allowed me to run the scripts without the Server service running:

````
@sfc 2>&1 | find /i "/SCANNOW" >nul
@IF %ERRORLEVEL% NEQ 0 GOTO ELEVATE
````

Someone should confirm whether this hack also works on Windows 8 and Windows 10.

For more information, see https://stackoverflow.com/a/21295806/1650137 (and the other answers as well).

**Comment by Biggizen** [comment]
I can confirm the loop. It's easily stopable with alt + F4. This isn't meant as solution, I only want to give this hint, so that no one gets panicky, when this "DOS-attack" (lol) goes off ;)

**Comment by jedisct1** [comment]
The change has been made. Thanks! :)

**Comment by a1346054** [comment]
this also fixes #193
